bundles/admin/admin-layereditor/view/AdminLayerForm/PermissionTabPane/PermissionRow.jsx

```diff
@@ -0,0 +1,81 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import { Checkbox } from 'oskari-ui';
+
+const RowContainer = styled.div`
+    display: flex;
+    flex-wrap: wrap;
+`;
+
+const TextColumn = styled.div`
+    flex-grow: 1;
+    width: 195px;
+`;
+
+const HeaderTextColumn = styled.div`
+    flex-grow: 1;
+    width: 195px;
+    align-self: flex-end;
+`;
+
+const StyledPermissionDiv = styled.div`
+    flex-grow: 1;
+    float: left;
+    width: 110px;
+    display: flex;
+    flex-wrap: wrap;
+    justify-content: center;
+    padding: 5px !important;
+`;
+
+const Break = styled.div`
+    flex-basis: 100%;
+    height: 0;
+  
+`;
+
+const HeaderPermissionText = styled.div`
+    height: 80%;
+    text-align: center;
+    padding-bottom: 5px;
+`;
+
+const SelectAllDiv = styled.div`
+    height: 20%;
+    padding-bottom: 5px;
+`;
+
+export const PermissionRow = (props) => {
+    const permissionTypes = [];
+    props.permissionTypes.forEach(permission => {
+        if (props.isHeaderRow) {
+            permissionTypes.push(<StyledPermissionDiv key={permission.id}><HeaderPermissionText>{permission.selectionText}</HeaderPermissionText><Break/><SelectAllDiv><Checkbox/></SelectAllDiv></StyledPermissionDiv>);
+        } else {
+            permissionTypes.push(<StyledPermissionDiv key={permission.id}><Checkbox/></StyledPermissionDiv>);
+        }
+    });
+
+    return (
+        props.isHeaderRow
+            ? <RowContainer>
+                <HeaderTextColumn>
+                    { props.rowText}
+                </HeaderTextColumn>
+                { permissionTypes }
+            </RowContainer>
+            : <RowContainer>
+                <TextColumn>
+                    { props.rowText}
+                </TextColumn>
+                { permissionTypes }
+            </RowContainer>
```

Comment:
How about doing two returns here to make this more readable?

  if (props.isHeaderRow) {
     return (...);
  }
  return (...);

Reply:
Or consider doing two exports. One for header row and another for normal one. Then you could take out the isHeaderRow prop.

Reply:
I ended up doing conditional styling in TextColumn styled.div so HeaderTextColumn is removed as obsolete and if statement is no longer needed in render

Reply:
With refactoring done in latest commits, code is still work in progress in scope of whole layer permissions handling but it should be in better shape to add actual state and saving logic to permissions later on in other pull request(s).